### ISSUE TITLE
`st_intersection` failing to return intersection point between two linestrings using ESRI geometry engine

### ISSUE THREAD

**Issue by tompeterken-os** [databrickslabs/mosaic#299]
**Describe the bug**
When using ESRI geometry bindings, `st_intersection` applied to two intersecting LineStrings returns an empty MultiLineString rather than a Point geometry, even when `st_intersects` returns True.

When using JTS bindings, `st_intersection` returns a Point geometry as expected.

**To Reproduce**
I've generated a minimal example below:
```
import json

# Construct two lines which intersect at (1,1)
vertical_line = {"type":"LineString","coordinates":[[1,0], [1,2]]}
horizontal_line = {"type": "LineString", "coordinates": [[0,1], [2,1]]}

# Create a DataFrame
df = (
  spark.createDataFrame([{'vertical': json.dumps(vertical_line), 'horizontal': json.dumps(horizontal_line)}])
  
  # st_intersects should return True
  .withColumn(
    'intersects',
    mos.st_intersects(mos.st_geomfromgeojson('vertical'), mos.st_geomfromgeojson('horizontal'))
  )
  
  # st_intersection should return a point geometry with coords (1,1)
  .withColumn(
    'intersection_geometry',
    mos.st_asgeojson(mos.st_intersection(mos.st_geomfromgeojson('vertical'), mos.st_geomfromgeojson('horizontal')))
  )
)

display(df)
```

**Expected behavior**
The `intersection_geometry` column in `df` above should return a Point geometry with coordinates (1,1).

**Screenshots**
Result with default ESRI bindings:
![image](https://user-images.githubusercontent.com/88335155/219720645-1b762812-06ce-42e0-92d7-0eededc85148.png)

Result with JTS bindings, showing expected behaviour:
![image](https://user-images.githubusercontent.com/88335155/219721817-7c83c56d-5721-4aa0-af89-7f7332cdc828.png)

**Additional context**
I'm using mosaic 0.3.7 with Databricks Runtime 11.3 LTS ML, Apache Spark 3.3.0, Scala 2.12.

**Comment by edurdevic** [comment]
Thank you for reporting this @tompeterken-os.
We are considering to deprecate the ESRI bindings and switch the default to JTS, because there are a few bugs like this one that we can't easily fix with ESRI.

**Comment by mickvanhulst-TomTom** [comment]
@edurdevic maybe good to know that JTS also has issues related to intersection. At times, I've received `found non-noded intersection` exceptions when tesselating line geometries using mosaic or by using `st_intersection` directly. 

https://locationtech.github.io/jts/jts-faq.html#D1

For some other use-case, I resolved this by using `GeometryClipper`, but this only works with rectangles. 

https://docs.geotools.org/latest/javadocs/org/geotools/geometry/jts/GeometryClipper.html

**Comment by tompeterken-os** [comment]
Fixed in version 0.3.10, thank you @milos-colic and team!
